devices/Sim7080/Sim7080G.cs

```diff
@@ -0,0 +1,413 @@
+        /// </summary>
+        public void GetNetworkSystemMode()
+        {
+            SimController.GetSystemMode(_serialPort);
+        }
+
+        /// <summary>
+        /// Set the network system mode for wireless and cellular network communication service.
+        /// </summary>
+        /// <param name="systemMode">The desired <see cref="SystemMode"/>.</param>
+        /// <param name="enableReporting">Report the network system mode information.</param>
+        /// <param name="wait">The time to wait to switch system mode.</param>
+        public void SetNetworkSystemMode(SystemMode systemMode = SystemMode.GSM, bool enableReporting = true, int wait = 5000)
+        {
+            SimController.SetSystemMode(_serialPort, systemMode, enableReporting);
+
+            Thread.Sleep(wait);
+        }
+
+        /// <summary>
+        /// Connect to the cellular network.
+        /// </summary>
+        /// <param name="apn">Cellular network access point name.</param>
+        /// <returns><see cref="ConnectionStatus"/></returns>
+        public ConnectionStatus NetworkConnect(string apn)
+        {
+            if (!_serialPort.IsOpen)
+            {
+                NetworkInformation.ConnectionStatus = ConnectionStatus.Disconnected;
+                return NetworkInformation.ConnectionStatus;
+            }
+
+            var retryCount = 0;
+
+            do
+            {
+                retryCount++;
+
+                SimController.NetworkConnect(_serialPort, apn);
+            }
+            while (NetworkInformation.ConnectionStatus == ConnectionStatus.Disconnected && retryCount < Retry);
+            return NetworkInformation.ConnectionStatus;
+        }
+
+        /// <summary>
+        /// Disconnect to the cellular network .
+        /// </summary>
+        /// <returns><see cref="ConnectionStatus"/></returns>
+        public ConnectionStatus NetworkDisconnect()
+        {
+            if (_serialPort.IsOpen &&
+                NetworkInformation.ConnectionStatus == ConnectionStatus.Connected)
+            {
+                SimController.NetworkDisconnect(_serialPort);
+            }
+
+            return NetworkInformation.ConnectionStatus;
+        }
+
+        /// <summary>
+        /// Connect to Azure IoT Hub using MQTT protocol.
+        /// </summary>
+        /// <param name="deviceId">The device identifier.</param>
+        /// <param name="hubName">The Azure IoT Hub Name.</param>
+        /// <param name="sasToken">The Secure Access Token.</param>
+        /// <param name="portNumber">The MQTT port number.</param>
+        /// <param name="apiVersion">The Azure IoT Hub API version.</param>
+        /// <param name="wait">The time to wait to establish the connection.</param>
+        /// <returns><see cref="ConnectionStatus"/></returns>
+        public ConnectionStatus ConnectAzureIoTHub(string deviceId, string hubName, string sasToken, int portNumber = 8883, string apiVersion = "2021-04-12", int wait = 5000)
```

Comment:
Can you bring out of this file, in another class anything related to Azure? It will make things really more clear and more maintainable.